api/src/routers/meals.js

```diff
@@ -79,4 +81,129 @@ mealsRouter.get("/last", async (req, res) => {
   }
 });
 
+mealsRouter.get("/:id", async (req, res) => {
+  try {
+    const id = Number(req.params.id);
+    const meal = await knex("meals").where("id", "=", id).first();
+
+    if (isNaN(id)) {
+      res.status(400).json({ error: "Please provide a meal ID number." });
+    }
+
+    if (!meal) {
+      res.status(404).json({ error: "Meal not found." });
+    }
+
+    res.json(meal);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ error: "An error ocurred while trying to fetch data." });
+  }
+});
+
+mealsRouter.post("/", async (req, res) => {
+  try {
+    const {
+      title,
+      description,
+      location,
+      when,
+      max_reservations,
+      price,
+      created_date,
+    } = req.body;
+
+    if (
+      !title ||
```

Comment:
You could simplify this by and-ing all the variables and then negating the result, i.e.
`if ( !( title && description && ... ) )`

But of course either way is totally correct.